### PR TITLE
Only allow navigation to a folder if you have access

### LINKED ISSUE #18716: [others with access] Sharee can click link to unaccessible level
image

### Steps to reproduce
1. Henri is the owner of _Clubs_
2. Henri shares _Clubs_ with Peter (with resharing permissions)
3. Henri shares _Kreuzberg_ with Simon (with resharing permissions)
4. Peter and Simon accept the shares
5. Folder _Clubs_ appears on Peter's root level
6. Folder _Kreuzberg_ appears on Simon's root level
7. Peter opens the list of _others with access_ on _Lido_
8. Peter sees Simon in the list
9. Peter clicks on "Via folder" in the ... menu behind Simon
10. Folder _Kreuzberg_ opens
11. Simon opens the list of _others with access_ on _Lido_
12. Simon sees Peter in the list

### Expected behaviour
- Simon sees Peter in the list of _others with access_
- In the ... menu behind Peter, Simon sees only "Added by Henri"

### Actual behaviour
- In the ... menu behind Peter, Simon also sees "Via folder"
- Click on "Via folder" navigates Simon to a folder ID Simon has no access rights to

image

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #18716

If you can delete the share then you have access yourself to that shares
location.

@wiswedel please double check your use cases ;)

### CODE REVIEW

**Comment by wiswedel** [comment]
It's not 100% ideal since you may have access anyway) but better than getting errors when you have too little access.

File: apps/files_sharing/src/components/SharingEntryInherited.vue [hunk 1]
```diff
@@ -35,7 +35,7 @@
 		<ActionText icon="icon-user">
 			{{ t('files_sharing', 'Added by {initiator}', { initiator: share.ownerDisplayName }) }}
 		</ActionText>
-		<ActionLink v-if="share.fileSource"
+		<ActionLink v-if="share.canDelete && share.fileSource"
```
Comment (wiswedel) [inline apps/files_sharing/src/components/SharingEntryInherited.vue]: Is ` && share.fileSource` even needed anymore?

Comment (skjnldsv) [reply apps/files_sharing/src/components/SharingEntryInherited.vue]: Well, if there is no file source, it doesn't make sense to show this entry :)

Comment (skjnldsv) [reply apps/files_sharing/src/components/SharingEntryInherited.vue]: (unlikely, but I've seen unlikelier thing happen :wink: )

Comment (wiswedel) [reply apps/files_sharing/src/components/SharingEntryInherited.vue]: > Well, if there is no file source, it doesn't make sense to show this entry :)

When can there be no file source?

Comment (rullzer) [reply apps/files_sharing/src/components/SharingEntryInherited.vue]: It is just code hardening ;)

**Review by juliusknorr — APPROVED** [review]

**Pull request merged**